pkg/interface/link/src/js/api.js

```diff
@@ -91,135 +91,84 @@ class UrbitApi {
     });
   }
 
-  async getComments(path, url, page, index) {
-    let endpoint = "/~link/discussions" + path + "/" + window.btoa(url) + ".json?p=0";
-    let promise = await fetch(endpoint);
-    if (promise.ok) {
-      let comments = {
-        "link-update": {
-          comments: {
-            path: path,
-            page: page,
-            index: index
-          }
-        }
-      };
-      comments["link-update"].comments.data = await promise.json();
-      store.handleEvent(comments);
-    }
+  getComments(path, url) {
+    return this.getCommentsPage.bind(this)(path, url, 0);
   }
 
-  async getCommentsPage(path, url, page, index, commentPage) {
-    let endpoint = "/~link/discussions" + path + "/" + window.btoa(url) + ".json?p=" + commentPage;
-    let promise = await fetch(endpoint);
-    if (promise.ok) {
-      let responseData = await promise.json();
-      let update = {
-        "link-update": {
-          commentPage: {
-            path: path,
-            linkPage: page,
-            index: index,
-            comPageNo: commentPage,
-            data: responseData.page
+  getCommentsPage(path, url, page) {
+    //TODO factor out
+    // encode the url into @ta-safe format, using logic from +wood
```

Comment:
I know it's here as a TODO, but yeah this would be good as a utility function in `/lib/util.js`